pkg/schemadsl/generator/generator_test.go

```diff
@@ -124,7 +124,10 @@ func TestGenerator(t *testing.T) {
 				),
 				namespace.RelationWithComment("reader", "//foobar", nil,
 					namespace.AllowedRelation("foos/user", "..."),
+					namespace.AllowedPublicNamespace("foos/user"),
 					namespace.AllowedRelation("foos/group", "member"),
+					namespace.AllowedRelationWithCaveat("foos/user", "...", namespace.Caveat("somecaveat")),
+					namespace.AllowedRelationWithCaveat("foos/group", "member", namespace.Caveat("somecaveat")),
```

Comment:
we are missing public namespace with caveat

Reply:
Good catch; fixed